### PR TITLE
add an example with web3 and nextauth.js

### PR DESCRIPTION
**Pull request by gbibeaul**
### Description

Often applications using Metamask to authenticate users resort to fetchig everything from the client as Metamask works by injecting itself on the window object. This means these apps are missing a lot of what Next.js has to offer. The techniques in this example allow users to get their connected users wallet and use them with great Next.js features

- `middleware`
- `getServerSideProps`
- Eventually RSC

[Readme of the example](https://github.com/vercel/examples/tree/feat/web3-nextauthjs/solutions/web3-sessions)

### Best Way to Test

1. Get the metamask extension and setup a wallet
2. Try navigating to `/protected` you should be redirected to `/`
3. Login, authorize the app in metamask
4. On `/protected` try switching to a new sub wallet or forcing a metamask disconnect. You will be redirected to `/`
5. try the regular logout 

### Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New Example
- [ ] New feature in existing example

### New Example Checklist

- [x] 🚀 Link to deployment URL on Vercel
- [x] 📱 Is it responsive? Are mobile and tablets considered?
- [x] 📚 `README.md` preview link in PR description (branch)
- [x] ⚙️ Secrets have instructions for how to set up in the readme
- [x] 🛫 `npm run new-example` was run to create the example

### CODE REVIEW

File: solutions/web3-sessions/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+This Demo will show how to configure NextAuth.js and create a custom hook to protect routes.
+
+## How to Use
+
+You will need to add the proper variables in a `.env` file like show in the `.env.example` file.
+
+```
+// generate random secrets on https://passwordsgenerator.net/ or anywhere else similar
+NEXT_AUTH_SECRET=123
+JWT_SECRET=123
```
Comment (lfades) [inline solutions/web3-sessions/README.md:20]: Let's not use `123` as the example value here but a long and secure password instead.

File: solutions/web3-sessions/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# web3-sessions example
```
Comment (lfades) [inline solutions/web3-sessions/README.md:1]:
```suggestion
# Web3 Sessions
```

File: solutions/web3-sessions/pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    }
+    const { data } = await connect(connectData.connectors[0])
+
+    signIn('credentials', { address: data?.account, callbackUrl })
+  }
+
+  return (
+    <Page>
+      <Head>
+        <title>Web3 Session Example</title>
```
Comment (lfades) [inline solutions/web3-sessions/pages/index.tsx:26]:
```suggestion
        <title>Web3 Session with NextAuth.js</title>
```

File: solutions/web3-sessions/pages/index.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        <meta
+          name="description"
+          content="Vercel example how to use web3-sessions"
```
Comment (lfades) [inline solutions/web3-sessions/pages/index.tsx:29]:
```suggestion
          content="Learn how to do Web3 sessions with NextAuth.js"
```

File: solutions/web3-sessions/pages/index.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        />
+        <link rel="icon" href="/favicon.ico" />
+      </Head>
+
+      <section className="flex flex-col space-y-4 gap-6">
+        <Text variant="h1">Web3 Session Example</Text>
```
Comment (lfades) [inline solutions/web3-sessions/pages/index.tsx:35]:
```suggestion
        <Text variant="h1">Web3 Session with NextAuth.js</Text>
```

File: solutions/web3-sessions/pages/index.tsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        <Text>
+          In a decentralized application, a user is often identifies by a
```
Comment (lfades) [inline solutions/web3-sessions/pages/index.tsx:37]:
```suggestion
          In a decentralized application, a user is often identified by a
```

File: solutions/web3-sessions/pages/index.tsx [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+          Cryptocurrency wallet such as{' '}
+          <Link href="https://metamask.io/" target="_blank">
+            Metamask
+          </Link>
+          . However, since Metamask works by injecting a script into the page,
+          it is only available on the client, cutting off the ability to use
+          `getServerSideProps` to fetch user data.
```
Comment (lfades) [inline solutions/web3-sessions/pages/index.tsx:44]: Use the `<Code>` component for `getServerSideProps` here

File: solutions/web3-sessions/pages/index.tsx [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+          We can solve this by pairing a{' '}
+          <Link href="https://next-auth.js.org/" target="_blank">
+            NextAuth.js
+          </Link>{' '}
+          session with a convenient hooks library called{' '}
+          <Link href="https://github.com/tmm/wagmi" target="_blank">
+            WAGMI
+          </Link>
+          . We will need to configure NextAuth.js with the
+          `CredentialsProvider`:
```
Comment (lfades) [inline solutions/web3-sessions/pages/index.tsx:56]: Use the `<Code>` component here

**Review by lfades — CHANGES_REQUESTED** [review]
Added some suggestion, as a side note I think the example should be called `web3-ssr-sessions` for the folder name and `Web3 SSR Sessions with NextAuth.js` for the title name, because it's mainly focused on SSR and the solution is NextAuth.js.

File: solutions/web3-sessions/README.md (hunk 1 above)
Comment (gbibeaul) [reply solutions/web3-sessions/README.md]: Good point!

**Review by gbibeaul — COMMENTED** [review]
@lfades Thank you for the review! I solved everything

**Review by lfades — APPROVED** [review]
Added some minor changes and it now looks good to me. Thank you!

**Pull request merged**